da/da.go

```diff
@@ -33,12 +34,12 @@ type BaseResult struct {
 	DAHeight uint64
 }
 
-// ResultSubmitBlock contains information returned from DA layer after block submission.
+// ResultSubmitBlock contains information returned from DA layer after block header/data submission.
 type ResultSubmitBlock struct {
 	BaseResult
 	// Not sure if this needs to be bubbled up to other
 	// parts of Rollkit.
-	// Hash hash.Hash
+	Hash header.Hash
```

Comment:
this is temporary solution, as we discussed earlier, until we have the blob module to help us compute the DA commitment, we use simply hash(block.Data) to assign to header.DataHash and sign it before submitting the signed header to DA.
ResultSubmitBlock is locally populated by the DA client (e.g., celestia.go) based on the result obtained from the DA call (mostly tx hash) and it does not have wait. Meaning, we could compute and populate the DA commitment and return as part of ResultSubmitBlock after the SubmitBlockData call without introducing additional wait, so that we can assign to header.DataHash and proceed to SubmitBlockHeader to DA.

Reply:
we can adjust the logic after the blob module is available?

Reply:
I would adjust it now as this changes the behaviour in a direction that does not need to change. We are introducing another Blocktime of waiting which could be avoided now.